packages/pluginutils/src/createFilter.ts

```diff
@@ -0,0 +1,54 @@
+import { resolve, sep } from 'path';
+
+import mm from 'micromatch';
+
+import { CreateFilter } from '../types';
+
+import ensureArray from './utils/ensureArray';
+
+function getMatcherString(id: string, resolutionBase: string | false | null | undefined) {
+  if (resolutionBase === false) {
+    return id;
+  }
+  return resolve(...(typeof resolutionBase === 'string' ? [resolutionBase, id] : [id]));
+}
+
+const createFilter: CreateFilter = function createFilter(include?, exclude?, options?) {
+  const resolutionBase = options && options.resolve;
+
+  const getMatcher = (id: string | RegExp) =>
+    id instanceof RegExp
+      ? id
+      : {
+          test: mm.matcher(
+            getMatcherString(id, resolutionBase)
+              .split(sep)
+              .join('/'),
+            { dot: true }
+          )
+        };
+
+  const includeMatchers = ensureArray(include).map(getMatcher);
+  const excludeMatchers = ensureArray(exclude).map(getMatcher);
+
+  return function result(id: string | any): boolean {
```

Comment:
`unknown` has better type safety here, and the `typeof` check should ensure that it doesn't affect the other code.

```suggestion
  return function result(id: string | unknown): boolean {
```

Reply:
I'll make a note to circle back to this once migration is complete. That's coming from the old repo, so I'd like to leave it as-is for now.